src/rulesets/Base/array.jl

```diff
@@ -527,3 +527,42 @@ function _extrema_dims(x, dims)
     end
     return y, extrema_pullback_dims
 end
+
+#####
+##### `pmap`
+#####
+
+using UUIDs # of course we don't want this;
+            # what is a better way of uniquely identifying function calls from the forward pass?
+
+save_backs = Dict() # could this be a stack instead?
+
+# Now that there is a backwards rule for zip (albeit only in Zygote),
+# it should be fine to deal with only a single collection c
+function rrule(config::RuleConfig{>:HasReverseMode}, ::typeof(pmap), f, p::AbstractWorkerPool, c; kwargs...)
+    key = uuid1()
+    function forw(i, a)
+        y, back = rrule_via_ad(config, f, a)
+        save_backs[(key,i)] = back;
+        return y, myid()
+    end
+
+    I = reshape(eachindex(c), size(c))
```

Comment:
Are any non-array collections allowed? If not, it should probably be `::AbstractArray` in the signature. And `axes` not size here. 

Also, can I vote to call the big collection a bigger letter than `c`? Like A, of at least C?

Reply:
I was hoping to handle arbitrary collections (so perhaps `I = reshape(1:length(A), axes(A))`) instead), but that would complicate things in the pullback (I'm not sure how to express the tangent of an arbitrary collection, right now I'm just returning an array). So I'll add an `::AbstractArray` signature for now.